src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/collector/NodeConfigCollector.java

```diff
@@ -101,6 +105,13 @@ private void collectHeapStats(MetricFlowUnit heapMax) {
     collectAndPublishMetric(ResourceUtil.YOUNG_GEN_MAX_SIZE, edenMaxSize + (2 * survivorMaxSize));
   }
 
+  private void collectHeapUsageStats(MetricFlowUnit heapUsed) {
```

Comment:
How do we decide b/w putting something in the NodeConfigCollector (or NodeConfigCache) v/s creating a node level and cluster level rca for it? Both are ways to get values to the master node deciders.

Up until now, we were using NodeConfigCache only for configuration values passed as a metric reading - e.g. queue capacity, heap max size, cache max size. Heap usage is a digression from that pattern.
I'm not saying this is wrong, but we should have some structure/design so as to avoid making the NodeConfigCollector as one big blob of all metrics.

IMO, it makes sense to use NodeConfigCollector only for fixed config values (like we have up until now), and move things like usage metrics (heap usage) to node and cluster level RCAs. This leaves the door open for any future post-processing we might want, e.g. returning the avg/min/ewma of heap usage over a sliding window rather than exact last value.